jsettlers.logic/src/jsettlers/ai/highlevel/AiStatistics.java

```diff
@@ -0,0 +1,612 @@
+				if (landscapeGrid.getResourceAmountAt(x, y) > 0) {
+					if (!sortedResourceTypes.get(resourceType).containsKey(xInteger)) {
+						sortedResourceTypes.get(resourceType).put(xInteger, new ArrayList<Integer>());
+					}
+					sortedResourceTypes.get(resourceType).get(xInteger).add(yInteger);
+				}
+				Player player = partitionsGrid.getPlayerAt(x, y);
+				if (player == null) {
+					if (objectsGrid.hasCuttableObject(x, y, TREE_ADULT)) {
+						if (!sortedCuttableObjectsInDefaultPartition.containsKey(TREE_ADULT)) {
+							sortedCuttableObjectsInDefaultPartition.put(TREE_ADULT, new HashMap<Integer, List<Integer>>());
+						}
+						if (!sortedCuttableObjectsInDefaultPartition.get(TREE_ADULT).containsKey(xInteger)) {
+							sortedCuttableObjectsInDefaultPartition.get(TREE_ADULT).put(xInteger, new ArrayList<Integer>());
+						}
+						sortedCuttableObjectsInDefaultPartition.get(TREE_ADULT).get(xInteger).add(yInteger);
+					}
+					if (objectsGrid.hasCuttableObject(x, y, STONE)) {
+						if (!sortedCuttableObjectsInDefaultPartition.containsKey(STONE)) {
+							sortedCuttableObjectsInDefaultPartition.put(STONE, new HashMap<Integer, List<Integer>>());
+						}
+						if (!sortedCuttableObjectsInDefaultPartition.get(STONE).containsKey(xInteger)) {
+							sortedCuttableObjectsInDefaultPartition.get(STONE).put(xInteger, new ArrayList<Integer>());
+						}
+						sortedCuttableObjectsInDefaultPartition.get(STONE).get(xInteger).add(yInteger);
+					}
+					if (landscapeGrid.getLandscapeTypeAt(x, y) == RIVER1 || landscapeGrid.getLandscapeTypeAt(x, y) == RIVER2
+							|| landscapeGrid.getLandscapeTypeAt(x, y) == RIVER3 || landscapeGrid.getLandscapeTypeAt(x, y) == RIVER4) {
+						if (!sortedRiversInDefaultPartition.containsKey(xInteger)) {
+							sortedRiversInDefaultPartition.put(xInteger, new ArrayList<Integer>());
+						}
+						sortedRiversInDefaultPartition.get(xInteger).add(yInteger);
+					}
+				} else {
+					Integer playerId = new Integer(partitionsGrid.getPlayerAt(x, y).playerId);
+					ShortPoint2D point = new ShortPoint2D(x, y);
+					updateBorderlandNextToFreeLand(playerId, point);
+					if (!land.containsKey(playerId)) {
+						land.put(playerId, new ArrayList<ShortPoint2D>());
+					}
+					land.get(playerId).add(point);
+					if (!stones.containsKey(playerId)) {
+						stones.put(playerId, new ArrayList<ShortPoint2D>());
+					}
+					if (objectsGrid.hasCuttableObject(x, y, EMapObjectType.STONE)) {
+						stones.get(playerId).add(point);
+					}
+					if (!trees.containsKey(playerId)) {
+						trees.put(playerId, new ArrayList<ShortPoint2D>());
+					}
+					if (objectsGrid.hasCuttableObject(x, y, EMapObjectType.TREE_ADULT)) {
+						trees.get(playerId).add(point);
+					}
+					if (!rivers.containsKey(playerId)) {
+						rivers.put(playerId, new ArrayList<ShortPoint2D>());
+					}
+					if (landscapeGrid.getLandscapeTypeAt(x, y) == RIVER1 || landscapeGrid.getLandscapeTypeAt(x, y) == RIVER2
```

Comment:
Consider using:
RIVERS = EnumSet.of(RIVER1, RIVER2, ...);
RIVERS.contains(landscapeGrid.getLandscapeTypeAt(x, y))